blog: "Announcing Papyrs: Blog on web3" (#71)
peterpeterparker/daviddalbusco.com: src/blog/announcing-papyrs-blog-on-web3.md

date: "2022-06-08"
title: "Announcing Papyrs: Blog on web3"
description: "I have developed a new open-source, privacy-first, decentralized blogging platform that lives 100% on chain and today I am making it available to anyone."
tags: "#startup #programming #web3 #blockchain"
image: "https://miro.medium.com/max/1400/1*0aVC6yVZLehL990fUJDazw.png"
canonical: "https://medium.com/dfinity/dynamically-create-canister-smart-contracts-in-motoko-d3b38a748c07"
---
![https://6zvwc-sqaaa-aaaal-aalma-cai.raw.ic0.app/images/social-image.jpg?token=CLmDdhvrv_oiB27vvPeb9](https://6zvwc-sqaaa-aaaal-aalma-cai.raw.ic0.app/images/social-image.jpg?token=CLmDdhvrv_oiB27vvPeb9)

So, I have developed a new open-source, privacy-first, decentralized blogging platform that lives 100% on chain and today I am making it available to anyone. I called it: [Papyrs](https://app.papy.rs).

* * *

## What a ride!

Last year, I was lucky enough to receive grants from the DFINITY foundation to port our web editor for slides, [DeckDeckGo](https://deckdeckgo.com/), to the [Internet Computer](https://smartcontracts.org/).

While I completed all milestones, I did not migrated it yet.

> Along the way, I had the idea of extending the platform to support documents. One thing leading to another, I ended developing my own rich text editor library, [Stylo](https://stylojs.com/), and decided to focus first on this new format hence me launching today a new blogging platform.

Reviewer: ```suggestion
Along the way, I had the idea of extending the platform to support documents. One thing leading to another, I ended developing my own rich text editor library, [Stylo](https://stylojs.com/), and decided to focus first on this new format; which is why today I'm launching a new blogging platform.
```

> On the one hand, thanks to this approach, I limit the number of functions I launch on the IC, which allows me to gain experience to later deploy our presentation editor more serenely.

Author: Functions in the sense features. Papyrs has less features than DeckDeckGo therefore the scope is more narrowed - i.e. less risk of issues. Does it make sense?

> On the one hand, thanks to this approach, I limit the number of functions I launch on the IC, which allows me to gain experience to later deploy our presentation editor more serenely.

Reviewer: Ah! functionalities then I guess, but features is clearer

On the other hand, I like to develop new ideas and thus double the fun 😜.

* * *

## Manifesto

So, why launching another blogging application? After-all there are already many platforms that allow you to write articles isn't 🤷‍♂️?

Sure there is but none are following the guidelines that drive me.

* * *

### Decentralization & blockchain

The concept of decentralization - or web3 - is linked to a certain form of romanticism for developers that [Chris Dixon](https://twitter.com/cdixon) summarizes in following terms:

*The lesson is that when you compare centralized and decentralized systems you need to consider them dynamically, as processes, instead of statically, as rigid products. Centralized systems often start out fully baked, but only get better at the rate at which employees at the sponsoring company improve them. Decentralized systems start out half-baked but, under the right conditions, grow exponentially as they attract new contributors ([source](https://twg.io/whiteboard/why-decentralization-matters/)).*

> As a programmer my self, this idea is appealing. I also think that blockchain technology might lead to interesting new perspective for the future.

Reviewer: ```suggestion
As a programmer myself, this idea is appealing. I also think that blockchain technology might lead to interesting new perspectives for the future.
```

> I can notably imagine that consensus and other certification mechanism could potentially be useful to proof the validity of the articles that would be shared with Papyrs - i.e. could be used to prevent fake news and misinformation.

Reviewer: ```suggestion
I can for instance imagine that consensus and other certification mechanisms could potentially be useful to prove the validity of the articles shared with Papyrs - i.e. could be used to prevent fake news and misinformation.
```

But that is for the future. Right now, my main commitment is empowering users ownership of their data by taking advantages of smart contracts.

> Between ads networks - that I cannot stand - and Big Tech companies that exploit our private data, I developed a bit a pessimist view on the what has become Internet. Furthermore, as my parents say when they are pessimistic, it also feels that "the best is not to come".

Reviewer: ```suggestion
Between ads networks - that I cannot stand - and Big Tech companies that exploit our private data, I developed a bit a pessimist view on the what the Internet has become. Furthermore, as my parents say when they are pessimistic, it also feels that "the best is not to come".
```

That's why I developed Papyrs with a true privacy first approach.

* * *

### Privacy first

The editor works primarily offline and requires no sign-in. The blog posts can be loaded and exported to the file system. There is a Chrome [plugin](https://chrome.google.com/webstore/detail/papyrs-to-markdown/caacmbgdcjpdpmccocmbiljodkbkjglh) to convert these to markdown.

> If you are interested to save and share your blog posts online - on chain - then an authentication with [Internet Identity](https://identity.ic0.app/) is required.

Reviewer: ```suggestion
If you are interested in storing and sharing your blog posts online - on chain - then an authentication with [Internet Identity](https://identity.ic0.app/) is required.
```

> However, unlike any other auth providers, using this powerful password-less method preserve anonymity!

Reviewer: ```suggestion
However, unlike any other auth providers, using this powerful password-less method preserves anonymity!
```

Therefore, here are all the links to each and every single lines of code I wrote for Papyrs:

*   [Papyrs](https://github.com/papyrs/papyrs): the main repo - the web application
*   [IC](https://github.com/papyrs/ic): canisters and sign-in providers
*   [CDN](https://github.com/papyrs/cdn): libraries and assets used across the platform

*   [Unsplash-proxy](https://github.com/papyrs/unsplash-proxy): a proxy used to query Unsplash

*   [Stylo](https://github.com/papyrs/stylo): another kind of rich text editor

*   [Kit](https://github.com/papyrs/kit): the templates for the posts that are published on chain

*   [Papyrs to markdown](https://github.com/papyrs/markdown-plugin): a Chrome plugin to convert to markdown

*   [DeckDeckGo](https://github.com/deckgo/deckdeckgo/): various web components, utilities and the providers for the offline persistence and synchronization of the data

As the saying goes, "sharing is caring" 🤗. I hope what I implemented can help any developer starts building on the Internet Computer. Furthermore, I am also looking forward to feedbacks and contributions (🤞) in order to improve the project.

* * *

### Indie hacking

> Beside my opinionated point of views and the fact that I am the first user of my own project, I would lie if I would not admit that I have also some small hopes for Papyrs to generate revenue in the future.

Reviewer: ```suggestion
Beside my opinionated point of view and the fact that I am the first user of my own project, I would lie if I would not admit that I have also some small hopes for Papyrs to generate revenue in the future.
```

All this cannot happen without users but if some would be interested, I am convinced there will be a way to find a model that can be both profitable for Papyrs and the content creators while absolutely making no concession on my values.

* * *

## Features

> As mentioned in previous chapter, I use the platform for blogging purpose my self. That is why I designed it the way I want with the features I often need.

Reviewer: ```suggestion
As mentioned in the previous chapter, I use the platform for blogging purpose myself. That is why I designed it the way I want with the features I often need.
```

e.g. browsing [Unsplash](https://unsplash.com/) to select images for presentation purpose.

![https://unsplash.com/photos/uNNCs5kL70Q](https://images.unsplash.com/photo-1514984879728-be0aff75a6e8?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MnwzMDkyMzV8MHwxfHNlYXJjaHw0OXx8ZG9nJTIwaGVsbG98ZW58MHx8fHwxNjU0NDMzNDEy&ixlib=rb-1.2.1&q=80&w=1080)

*Photo by [Jamie Street](https://unsplash.com/@jamie452?utm_source=Papyrs&utm_medium=referral) on [Unsplash](https://unsplash.com/?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText)*

> Adding Gif from [Tenor](https://tenor.com/) for comic relief.

Reviewer: ```suggestion
Adding GIFs from [Tenor](https://tenor.com/) for comic relief.
```
